### ISSUE TITLE
Create API endpoint => projects/[id].ts/tasks.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#297]
Create the API endpoint projects/[id].ts/tasks.ts that will The tasks of the project detail API route. Be sure to add the proper extension when creating the file. Here are the instructions:

## Issue Description

This issue aims to provide a detailed description of the tasks required to create the `tasks.ts` API endpoint for the project detail route. The purpose of this endpoint is to fetch and manage tasks related to a specific project. The endpoint should support the following actions:

1. **GET**: Retrieve a list of tasks for a given project.
2. **POST**: Create a new task for a given project.
3. **PUT**: Update an existing task for a given project.
4. **DELETE**: Delete an existing task for a given project.

### Prerequisites

Before starting the implementation, ensure that the following prerequisites are met:

- The Prisma client is set up and available at `~/server/db.ts`.
- The relevant TypeScript types or Zod schemas are imported from the `types.ts` file.
- All data returned from a database query or external API call must be validated using Zod.

### Implementation Details

1. Create a new file at `~/pages/api/projects/[id]/tasks.ts` for the tasks API endpoint.
2. Import the required modules and types:

   ```typescript
   import { NextApiRequest, NextApiResponse } from "next";
   import { Task, TaskSchema } from "~/types";
   import { db } from "~/server/db";
   import { z } from "zod";
   ```

3. Define the `TasksResponse` TypeScript type for the expected output:

   ```typescript
   type TasksResponse = {
     tasks: Task[];
   };
   ```

4. Create the main API route handler function with the following signature:

   ```typescript
   export default async function handler(
     req: NextApiRequest,
     res: NextApiResponse<TasksResponse | Error>
   ): Promise<void> {
     // Implementation goes here
   }
   ```

5. Inside the `handler` function, use a switch statement to handle different request methods (GET, POST, PUT, DELETE):

   ```typescript
   switch (req.method) {
     case "GET":
       // Handle GET request
       break;
     case "POST":
       // Handle POST request
       break;
     case "PUT":
       // Handle PUT request
       break;
     case "DELETE":
       // Handle DELETE request
       break;
     default:
       res.setHeader("Allow", ["GET", "POST", "PUT", "DELETE"]);
       res.status(405).end(`Method ${req.method} Not Allowed`);
   }
   ```

6. Implement the GET request handler to fetch tasks for a given project:

   - Retrieve the project ID from the URL parameter.
   - Use the Prisma client to fetch tasks related to the project.
   - Validate the fetched tasks using the `TaskSchema` Zod schema.
   - Return the tasks in the response.

7. Implement the POST request handler to create a new task for a given project:

   - Retrieve the project ID from the URL parameter.
   - Parse and validate the request body using the `TaskSchema` Zod schema.
   - Use the Prisma client to create a new task with the validated data.
   - Return the created task in the response.

8. Implement the PUT request handler to update an existing task for a given project:

   - Retrieve the project ID and task ID from the URL parameters.
   - Parse and validate the request body using the `TaskSchema` Zod schema.
   - Use the Prisma client to update the task with the validated data.
   - Return the updated task in the response.

9. Implement the DELETE request handler to delete an existing task for a given project:

   - Retrieve the project ID and task ID from the URL parameters.
   - Use the Prisma client to delete the task.
   - Return a success message in the response.

### Error Handling

For each request handler, ensure proper error handling is in place. This includes:

- Handling invalid URL parameters or request body data.
- Catching any errors thrown by the Prisma client or Zod validation.
- Returning appropriate HTTP status codes and error messages in the response.

### Testing

Once the implementation is complete, test the API endpoint using a tool like Postman or by integrating it with the frontend application. Ensure that all actions (GET, POST, PUT, DELETE) work as expected and that proper error handling is in place.
